extensions/azure-app-config-client/runtime/src/main/java/io/quarkus/azure/app/config/client/runtime/DefaultAzureAppConfigClientGateway.java

```diff
@@ -0,0 +1,202 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.codec.digest.HmacAlgorithms;
+import org.apache.commons.codec.digest.HmacUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpEntityEnclosingRequest;
+import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScheme;
+import org.apache.http.client.AuthCache;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+class DefaultAzureAppConfigClientGateway implements AzureAppConfigClientGateway {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+    private final AzureAppConfigClientConfig azureAppConfigClientConfig;
+    private final URI baseURI;
+    private final String credential;
+    private final String secret;
+
+    public DefaultAzureAppConfigClientGateway(AzureAppConfigClientConfig azureAppConfigClientConfig) {
+        this.azureAppConfigClientConfig = azureAppConfigClientConfig;
+        try {
+            this.baseURI = determineBaseUri(azureAppConfigClientConfig);
+            this.credential = azureAppConfigClientConfig.credential;
+            this.secret = azureAppConfigClientConfig.secret;
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Value: '" + azureAppConfigClientConfig.url
+                    + "' of property 'quarkus.azure-app-config.url' is invalid", e);
+        }
+    }
+
+    private URI determineBaseUri(AzureAppConfigClientConfig azureAppConfigClientConfig) throws URISyntaxException {
+        String url = azureAppConfigClientConfig.url;
+        if (null == url || url.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "The 'quarkus.azure-app-config.name' property cannot be empty");
+        }
+        if (url.endsWith("/")) {
+            return new URI(url.substring(0, url.length() - 1));
+        }
+        return new URI(url);
+    }
+
+    @Override
+    public Response exchange() throws Exception {
+        final RequestConfig requestConfig = RequestConfig.custom()
+                .setConnectionRequestTimeout((int) azureAppConfigClientConfig.connectionTimeout.toMillis())
+                .setSocketTimeout((int) azureAppConfigClientConfig.readTimeout.toMillis())
+                //proxy?
```

Comment:
If we are going to support proxy, then please add the code. Otherwise please remove the field from `AzureAppConfigClientConfig`